bin/metrics-zookeeper-cluster.rb

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env ruby
+#
+# metrics-zookeeper.rb
+#
+# Collect ZooKeeper metrics
+# ===
+#
+# DESCRIPTION:
+# This plugin gathers metrics from an Exhibitor run ZooKeeper cluster,
+# based on the collectd plugin:
+#
+#   https://github.com/Nextdoor/collectd_plugins/blob/master/zookeeper/zookeeper.sh
+#
+#
+# PLATFORMS:
+#   Linux, BSD, Solaris
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#
+# LICENSE:
+# Sean Clemmer sczizzo@gmail.com
+# Released under the same terms as Sensu (the MIT license); see LICENSE
+#  for details.
+#
+
+require 'sensu-plugin/metric/cli'
+require 'socket'
+require 'net/http'
+require 'json'
+
+class ZookeeperMetrics < Sensu::Plugin::Metric::CLI::Graphite
+  option :exhibitor,
+         description: 'exhibitor end node for status checks',
+         short: '-e Exhibitor status end point',
+         long: '--exhibitor status end point',
+         default: 'http://localhost/exhibitor/v1/cluster/status'
+
+  option :scheme,
+         description: 'Metric naming scheme, text to prepend to metrics',
+         long: '--scheme SCHEME',
+         default: 'zookeeper.'
+
+  def dotted(*args)
+    args.join('.')
+  end
+
+  def zk_command(four_letter_word, host, port)
+    Socket.tcp(host, port) do |sock|
+      sock.print "#{four_letter_word}\r\n"
+      sock.close_write
+      sock.read
+    end
+  end
+
+  def exhibitor_status
+    response = json = ''
+    url = URI.parse(config[:exhibitor])
+    req = Net::HTTP::Get.new(url.path)
+    Net::HTTP.new(url.host, url.port).start do |http|
+      response = http.request(req)
+    end
+    return [false, json, ['exhibitor status is not http 200']] unless
```

Comment:
I think this could be written a bit better like this:
```
if response.is_a? Net::HTTPSuccess
  JSON.parse(response.body)
else
  [false, json, ['exhibitor status is not http 200']]
end
```